llvm/lib/Target/AMDGPU/SIRegisterInfo.cpp

```diff
@@ -1808,6 +1808,8 @@ void SIRegisterInfo::buildSpillLoadStore(
                            .addReg(SubReg, getKillRegState(IsKill));
         if (NeedSuperRegDef)
           AccRead.addReg(ValueReg, RegState::ImplicitDefine);
+        if (NeedSuperRegImpOperand && (IsFirstSubReg || IsLastSubReg))
```

Comment:
Why only first and last subreg?

Reply:
This is based on [D141101](https://reviews.llvm.org/D141101).
By marking implicit to only first and last, we will free up some unneeded dependencies for the post-ra scheduler.
CC @jrbyrnes

Reply:
copyPhysReg and the spill expansions should have similar behavior. I don't know why the spill case is more complex here

Reply:
If adding the implicit operand to all the subregs is the right approach (as currently happening in the copyPhysReg), we should fix all the paths in this function to include the same for all sub-register spill instructions.

Reply:
> marking implicit to only first and last spill instrs will free up some unneeded dependencies for the post-ra scheduler

Yes by reducing the implicit / implicit-def noise, we will reduce the dependencies in post-ra scheduling, which will allow the scheduler more flexibility for better scheduling.

Not to mention that these operands confuse the calls to MI.modifies/defines/reads Register -- by increasing the amount of implicit / implicit-def operands I assume we are just increasing our exposure to this. 

> copyPhysReg and the spill expansions should have similar behavior

I'm not sure either why I didn't make these changes to the copy lowering, but I agree it should behave the same. 

I think we are probably going to continue running into problems for these implicit / implicit-def operands until we can use liveness queries when doing the lowering. That said, I'm fine with adding back the implicit operands if that works best with the calls to modifies/defines/reads Register

Reply:
If we are keen on having the same behavior as `copyPhysReg`, how about submitting this patch first so that we will have same behavior in both the code paths (Spilling to VGPR + Spilling to Stack through VGPR) in this API? and then removing the conditional code which constraints the implicit for first and last spill instruction in the next patch?

Reply:
I would get this patch done first as we are not aggravating the complexity related to spill handling further. By adding implicit operands to the first and last subreg insns, this patch aligns the partial spill case (partially in VGPR and the rest into memory) to other vector spill cases handled in this function. Pravin can continue work on a follow-up patch to align it similar to the COPY expansion.